chall08/vsempere.py

```diff
@@ -0,0 +1,124 @@
+#! /usr/local/bin/python3
+
+import sys
+
+def process_user_input(user_input):
+    if (not isinstance(user_input, list)):
+        return None, None, None, 'Invalid input.';
+
+    map_line_size = -1
+    map_read = 0
+    moves_read = 0
+    map = ""
+    for line in user_input:
+        line = line.rstrip()
+        if (map_read == 0):
+            if (len(line) == 0):
+                map_read = 1
+            else:
+                if (map_line_size < 0):
+                    map_line_size = len(line)
+                elif (map_line_size != len(line)):
+                    return None, None, None, 'Invalid input.'
+                map = map + line
+        else:
+            if (moves_read == 0):
+                moves = line
+                moves_read = 1
+            else:
+                return None, None, None, 'Invalid input.'
+    return map, map_line_size, moves, None
```

Comment:
It seems to me like you are checking the input 2 times, you could have done everything in the same function!